adoc/extensions/sycl_khr_work_item_queries.adoc

```diff
@@ -0,0 +1,169 @@
+[[sec:khr-work-item-queries]]
+= SYCL_KHR_WORK_ITEM_QUERIES
+
+This extension allows developers to access instances of the
+[code]#sycl::nd_item#, [code]#sycl::group# and [code]#sycl::sub_group# classes
+without having to explicitly pass them as arguments to each function used on the
+device.
+
+{note} Passing such instances as arguments can result in a clearer interface
+that is less error-prone to use.
+For example, if a function accepts a [code]#sycl::group#, the caller must assume
+that function may call a [code]#sycl::group_barrier# and ensure that associated
+control flow requirements are satisfied.
```

Comment:
This sentence sounds too proscriptive for a non-normative note, especially the word "must".  How about:

> For example, when a function takes an argument of type `sycl::group`, it is an indication that the function may synchronize all work-items in that group by calling `sycl::group_barrier`.

Reply:
I thought that "must" would only be problematic here if we were speaking about something that an implementation must do.  I was trying to emphasize here that developers need to pay careful attention to APIs that accept group arguments .

I suppose what I really meant here was "developer" rather than "caller".  Would you still be opposed to wording like:

> For example, if a function accepts a `sycl::group`, developers must assume that function may call a `sycl::group_barrier` and ensure that associated control flow requirements are satisfied.

or:

> For example, if a function accepts a `sycl::group`, developers should assume that function may call a `sycl::group_barrier` and ensure that associated control flow requirements are satisfied.

I don't mind adopting even weaker wording if you still think this is problematic, but I wanted to check if my incorrect usage of "caller" here implied this was a note for implementers.

Reply:
I realized that you were addressing application developers with this note, but I still think the word "must" is wrong.  An application could write a function that takes a `group` parameter and still use an extension like root-groups to synchronize all work-items in the kernel.  We may think that's a bad style, but it is not prohibited.

Using the word "should" is better, but it still seems strong to me.  In the wording I proposed, I tried to point out the value of this style without badgering people to use it.

Reply:
Okay.  I've adopted your wording in 8cb242184a8594ed6bd038990101813a21eb7797.